src/DynamicData/List/ISourceList.cs

```diff
@@ -12,12 +13,36 @@ namespace DynamicData
     /// as well as data access methods
     /// </summary>
     /// <typeparam name="T"></typeparam>
-    public interface ISourceList<T> : IObservableList<T>
+    public interface ISourceList<T> : IObservableList<T>, IList<T>
```

Comment:
I am not sure whether it is desirable to fully implement `IList<T>`. I have always resisted doing so since I actively want to discourage people thinking that it should be treated in the same way as a normal list.

The reason  being that most list operations on source list are expensive in terms of inner locks and array copying. What I can be sure of is if consumers see Ilist then they will use it as a normal IList and this will surely lead to complaints about performance.

But then again maybe this is a needless worry of mine.

Reply:
I am interested in other opinions out there

Reply:
I'll just throw in my two cents, but I'm also interested in hearing some other's thoughts.

I agree with your sentiment that it's good to help guide people from using expensive objects in ways that feel cheap that might deceive them.  I'm totally on board with that up until it starts to hinder other important patterns and goals of users.  For me, the one that's being broken here is encouraging hefty amounts of duplicate code, which is a bigger architectural offense/concern for me.  Avoiding duplicate code heavily outweighs potentially helping newcomers avoid usage traps their first time around.

Not to mention, I think the familiar IList traps are all still there via the SourceListEditConvenienceEx class, even if they don't "officially" come from an IList implementation.  I know that I already personally have fallen into it not a few days ago, lol.  I was populating a SourceList, and using Add() one by one like an idiot, which was astronomically slow.  This is what made me create #277, as I opted to try out the BufferIf functionality rather than using AddRange.

Reply:
Having sat on this one for a day, and looking at the code again I am beginning to think that there may be no harm in implementing IList, after all as you point out most of the methods were available on the convenience methods.

However I think it wise for me to sit on this for a few days more as once an API it is available I have to support it forever.